### ISSUE TITLE
Incorrect no-missing-import error

### ISSUE THREAD

**Issue by mkay581** [runem/lit-analyzer#138]
Hello! 

We're running into an issue where the analyzer is incorrectly triggering a `no-missing-import` error. The file that is causing error is actually importing the proper file, but I believe its erroring because the file being imported doesn't have `customElements.define` in it. Instead, it has a custom function that we've abstracted to prevent from accidently double-defining customElements.

Any help on how to get around this?

Thanks for any help anyone can provide. Awesome package too, btw!

**Comment by 43081j** [comment]
from what i remember it will detect jsdoc, have you tried that?

```
/**
 * Some element
 * @customElement some-element
 */
class SomeElement extends HTMLElement {}
```

**Comment by mkay581** [comment]
No, I haven't.  That would require us to use jsdoc in every single one of our files. Unfortunately we can't do that.

**Comment by 43081j** [comment]
One way or another it'll be a code change on your end so I don't think you have a choice here. 

The analyzer will detect:

* `customElements.define`
* jsdoc `@customElement`
* Tag name maps (typescript global interface)
* Lit decorators `@customElement`

The analyzer has to detect your elements some way, right? So how do you think it can possibly do that if your classes contain no information to let us know they are elements?

**Comment by mkay581** [comment]
> So how do you think it can possibly do that if your classes contain no information to let us know they are elements?

:confused: Can you not detect that the classes extend `LitElement` or `HTMLElement`? I appreciate you trying to help but I could do without the sarcasm. Additionally, asking that a user use jsdoc (something that has nothing to do with this package) or suggesting that we change every single file in a project with thousands of files just for this single rule isn't helpful or unreasonable.

**Comment by 43081j** [comment]
It was only misread if it seemed like sarcasm. I was genuinely interested if you had something in mind as the existing architecture doesn't seem to have anything. 

In the ESLint lit plugin for example we do detect custom elements using the parent class (HTMLElement or a whitelist of class names). 

It's possible we could do similar here yes. But would probably bring some perf loss if we went beyond the immediate parent. A "dumb" check of "is the immediate parent class HTML element?" may work.

**Comment by 43081j** [comment]
we could change:
https://github.com/runem/web-component-analyzer/blob/master/src/analyze/flavors/custom-element/discover-definitions.ts

and discover direct subclasses of `HTMLElement` for example, possibly..
